### PR TITLE
Feature/1004 add version for kata monitor

### PR DESCRIPTION
**Pull request by liubin**
This PR will add `--version` to `kata-monitor` command, and print version info like this:

```
kata-monitor
 Version:	0.1.0
 Go version:	go1.14.6
 Git commit:	e1b3c90ac7e458d4c3b0ec69b3b374b1e7276472-dirty
 Built:	2020-10-21T02:11:26Z
 OS/Arch:	linux/amd64
```

And `data/kata-collect-data.sh` will collect this info.

Fixes: #1004

### CODE REVIEW

**Comment by fidencio** [comment]
@liubin, actually, having this backported to stable-2.0 branch would be good.

**Review by fidencio — APPROVED** [review]
lgtm, thanks @liubin!

**Review by jodh-intel — APPROVED** [review]
Thanks @liubin!

Random thought: although the versions details on this PR are nicely formatted, I wonder if it would be better to output a single line with all the details, more like the agent. We have a mix of formats so should probably try to make them all consistent (although that may break scripts, etc).

- Runtime:

  ```
  kata-runtime  : 2.0.0-rc0
     commit   : 8c3228c286278281f83377c1d60c27d95341919a-dirty
     OCI specs: 1.0.1-dev
  ```

- agent:

  ```
  kata-agent version 2.0.0-rc0 (api version: 0.0.1, commit version: 2.0.0-rc0-ce9a4eeffd0b5e6129085da3a4878c51c60b31dd-dirty, type: rust)
  ```

- The "ideal" (fully structured)?

  ```
  kata-monitor version="0.1.0", go-version="go1.14.6", git-commit="e1b3c90ac7e458d4c3b0ec69b3b374b1e7276472-dirty", build-date="2020-10-21T02:11:26Z" os="linux/amd64"
  ```

**Comment by jodh-intel** [comment]
Oh - one more comment - could you add a `func announce()` to log all the versions details like we do for other components (for example https://github.com/kata-containers/kata-containers/blob/2.0-dev/src/runtime/netmon/netmon.go#L242..L248).

**Comment by liubin** [comment]
@jodh-intel I have added `announce` log message.

About the output format(one line or multiple line),  multiple line is better for human reading. Key-value pair one line format is better for program to process. So we should print the version according to how does user use these commands.

File: src/runtime/Makefile [hunk 1]
```diff
@@ -36,6 +36,7 @@ ARCH_FILE_SUFFIX = -options.mk
 ARCH_FILE = $(ARCH_DIR)/$(ARCH)$(ARCH_FILE_SUFFIX)
 ARCH_FILES = $(wildcard arch/*$(ARCH_FILE_SUFFIX))
 ALL_ARCHES = $(patsubst $(ARCH_DIR)/%$(ARCH_FILE_SUFFIX),%,$(ARCH_FILES))
+BUILDTIME = $(shell date -u +"%Y-%m-%dT%H:%M:%SZ")
```
Comment (jodh-intel) [inline src/runtime/Makefile:39]: This is "docker-like", but it means the final binary isn't [reproducible](https://reproducible-builds.org/), so I'd question if we need this.

Comment (bergwolf) [reply src/runtime/Makefile:39]: +1, maybe a commit id and if it is dirty is more appropriate.

Comment (liubin) [reply src/runtime/Makefile]: Ok, I have deleted `Built` item.

**Pull request merged**